src/models/PagesEventDetails.ts

```diff
@@ -0,0 +1,78 @@
+import {PagesAnalyticsConfig} from './AnalyticsConfig';
+import {PagesEvent} from '../models';
+
+/**
+ * A fully detailed Pages Event
+ *
+ * @public
+ */
+export class PagesEventDetails implements PagesAnalyticsConfig, PagesEvent {
+  public businessId: number;
+  public eventType: string;
+  public siteId: number;
+  public production: boolean;
+  public ids?: number[];
+  public featureId: string;
+  public product: 'storepages' |'sites';
+  public path: string;
+  public pagesReferrer: string;
+  public pageType: 'entity'|'directory'|'locator'|'static';
+
+  constructor(config: PagesAnalyticsConfig, event: PagesEvent) {
+    this.eventType = event.eventType;
+    this.product = 'storepages';
+    this.siteId = config.siteId;
+    this.businessId = config.businessId;
+    this.production = config.production;
+    this.ids = config.ids;
+    this.pageType = config.pageType;
+
+    if (this.pageType === 'entity' && !this.ids) {
```

Comment:
If we want I believe this is enforceable at the type level. Do we also want to check the opposite, that entityIds aren't provided for non 'entity' page types?

Reply:
i'm not 100% certain. I believe they are optional, for example if you click on a result on a locator i think we can pass the entity id. I haven't had a chance to check

Reply:
ok i've confirmed that if you set the id currently it treats it as an entity page